client/src/components/Sub_RatingsReviews/generateStars.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import uuid4 from 'uuid4';
+import Star from './Star.jsx';
+
+export default function generateStars(rating, amtStars, size = 25) {
+  const result = [];
+  let remaining = rating;
+  let counter = 0;
+  while (counter < amtStars) {
+    if (remaining - 1 >= 0) {
+      remaining -= 1;
+      result.push(1);
+    } else if (remaining - 1 <= 0) {
+      result.push(remaining);
+      remaining = 0;
+    }
+    counter += 1;
```

Comment:
how about having a conditional like remaining remaining is greater than 0?

Reply:
and maybe remove counter?

Reply:
hey josh, if we were to stop the loop when remaining = 0, with nothing else unchanged, i don't think we'd be able to get empty, unfilled stars on the right side.

Reply:
maybe Array(amtStars).fill(0) for result and instead of pushing values and replace them?

Reply:
you're right, maybe that could be more useless work